OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatisticServiceTest.cs

```diff
@@ -18,68 +18,183 @@ namespace OutOfSchool.WebApi.Tests.Services
+        private Mock<IEntityRepository<Direction>> directionRepository;
+
         private Mock<ILogger> logger;
 
+        private IEnumerable<Workshop> workshops;
+        private IEnumerable<Application> applications;
+        private IEnumerable<Direction> directions;
+
         [SetUp]
         public void SetUp()
         {
-            context = new OutOfSchoolDbContext(UnitTestHelper.GetUnitTestDbOptions());
-            applicationRepository = new ApplicationRepository(context);
-            workshopRepository = new WorkshopRepository(context);
-            directionRepository = new EntityRepository<Direction>(context);
+            applicationRepository = new Mock<IApplicationRepository>();
+            workshopRepository = new Mock<IWorkshopRepository>();
+            directionRepository = new Mock<IEntityRepository<Direction>>();
 
             logger = new Mock<ILogger>();
-            service = new StatisticService(applicationRepository, workshopRepository, directionRepository, logger.Object);
+
+            service = new StatisticService(
+                applicationRepository.Object,
+                workshopRepository.Object,
+                directionRepository.Object,
+                logger.Object);
+
+            workshops = FakeWorkshops();
+            applications = FakeApplications();
+            directions = FakeDirections();
         }
 
         [Test]
         public async Task GetPopularWorkshops_ShouldReturnWorkshops()
         {
-            var directions = context.Directions;
-
             // Arrange
-            var expected = new WorkshopDTO
+            var expected = new List<WorkshopCard>
             {
-                Id = 1,
-                Title = "w1",
-                DirectionId = 1,
-                Direction = directions.First().Title,
-                Teachers = null,
-                Keywords = new List<string>() {string.Empty},
+                new WorkshopCard()
+                {
+                    WorkshopId = 3,
+                    Title = "w3",
+                },
+                new WorkshopCard()
+                {
+                    WorkshopId = 2,
+                    Title = "w2",
+                },
             };
 
+            var workshopsMock = workshops.AsQueryable().BuildMock();
+
+            workshopRepository.Setup(w => w.Get<It.IsAnyType>(
+                It.IsAny<int>(),
+                It.IsAny<int>(),
+                It.IsAny<string>(),
+                It.IsAny<Expression<Func<Workshop, bool>>>(),
+                It.IsAny<Expression<Func<Workshop, It.IsAnyType>>>(),
+                It.IsAny<bool>()))
+                .Returns(workshopsMock.Object);
+
             // Act
             var result = await service.GetPopularWorkshops(2).ConfigureAwait(false);
 
             // Assert
             result.Should().HaveCount(2);
-            result.Should().ContainEquivalentOf(expected);
+            result.Should().BeEquivalentTo(expected);
```

Comment:
You can use [CollectionAssert.AreEqual](https://docs.nunit.org/articles/nunit/writing-tests/assertions/classic-assertions/Collection-Assert.html)
It makes sure that both collections contains same items in the same order.

Reply:
Could I use `Should().BeEquivalentTo()` with option [WithStrictOrdering] (https://fluentassertions.com/objectgraphs/#ordering) instead? 
It is supposed to do the same: comparing lists members considering ordering.